### PR TITLE
Remove Hungarian notation from libnaucrates and libgpos

### PR DESCRIPTION
**Pull request by melanieplageman**
This is the first commit of the initiative to remove Hungarian notation
and replace it with a new convention -- a custom convention based on
common C++ naming conventions.
This PR only removes Hungarian notation from the libnaucrates and libgpos sub-directories of the GPORCA code base.

Corresponding GPDB WIP PR for translator changes: https://github.com/greenplum-db/gpdb/pull/5197

### CODE REVIEW

File: libgpos/include/gpos/common/CSyncPool.h [hunk 1]
```diff
@@ -43,248 +42,247 @@ namespace gpos
...
+				// object is reserved, check if it has been marked for recycling
+				if (bit_val == (bit_val & m_objs_recycled[elem_offset]))
+				{
+					// attempt to unset the recycle bit
+					if (UnsetBit(&m_objs_recycled[elem_offset], bit_val))
+					{
+#ifdef GPOS_DEBUG
+						BOOL recycled =
+#endif  // GPOS_DEBUG \
+	// unset the reserve bit - must succeed
```
Comment (d) [inline libgpos/include/gpos/common/CSyncPool.h]: I'm surprised this would compile

**Comment by d** [comment]
1. This patch doesn't seem to compile.

1. Can you also attach the corresponding Greenplum (mostly translator) change?

File: libgpos/include/gpos/common/CSyncPool.h (hunk 1 above)
Comment (khannaekta) [reply libgpos/include/gpos/common/CSyncPool.h]: Thanks @d. Updated and pushed the change.
Corresponding GPDB WIP PR for translator changes: https://github.com/greenplum-db/gpdb/pull/5197

**Comment by hsyuan** [comment]
happy to see Hungarian is gone.

File: libgpopt/include/gpopt/base/CStateMachine.h [hunk 2]
```diff
@@ -349,7 +349,7 @@ namespace gpopt
                 return fSucceeded;
             }
 			
-			// shorthand if current state and return value are not needed
+			// shorthand if current state and return m_bytearray_value are not needed
```
Comment (sambitesh) [inline libgpopt/include/gpopt/base/CStateMachine.h]: the word value has been replaced with m_bytearray_value in many places

File: libgpopt/include/gpopt/base/COptCtxt.h [hunk 3]
```diff
@@ -79,15 +79,15 @@ namespace gpopt
...
 
 			// optimizer configurations
-			COptimizerConfig *m_poconf;
+			COptimizerConfig *m_optimizer_config;
 
 			// whether or not we are optimizing a DML query
 			BOOL m_fDMLQuery;
 
-			// value for the first valid part id
+			// m_bytearray_value for the first valid part id
```
Comment (sambitesh) [inline libgpopt/include/gpopt/base/COptCtxt.h]: value??

File: libgpopt/include/gpopt/base/CColRefSet.h [hunk 4]
```diff
@@ -55,31 +55,31 @@ namespace gpopt
 		private:
 						
 			// determine if bit is set
-			BOOL FBit(ULONG ulBit) const;
+			BOOL Get(ULONG ulBit) const;
 			
-			// set given bit; return previous value
-			BOOL FExchangeSet(ULONG ulBit);
+			// set given bit; return previous m_bytearray_value
```
Comment (sambitesh) [inline libgpopt/include/gpopt/base/CColRefSet.h]: value?

File: libgpopt/include/gpopt/base/CColumnFactory.h [hunk 5]
```diff
@@ -50,10 +50,10 @@ namespace gpopt
 		private:
 
 			// MTS memory pool
-			IMemoryPool *m_pmp;
+			IMemoryPool *m_mp;
 
 			// mapping between column id of computed column and a set of used column references
-			HMCrCrs *m_phmcrcrs;
+			ColRefToColRefSetMap *m_phmcrcrs;
```
Comment (sambitesh) [inline libgpopt/include/gpopt/base/CColumnFactory.h]: should we also rename m_phmcrcrs?

File: libgpopt/include/gpopt/base/CConstraint.h [hunk 6]
```diff
@@ -249,26 +249,26 @@ namespace gpopt
...
 
 			// create disjunction from array of constraints
 			static
-			CConstraint *PcnstrDisjunction(IMemoryPool *pmp, DrgPcnstr *pdrgpcnstr);
+			CConstraint *PcnstrDisjunction(IMemoryPool *mp, CConstraintArray *pdrgpcnstr);
 
 			// merge equivalence classes coming from children of a bool op
 			static
-			DrgPcrs *PdrgpcrsMergeFromBoolOp(IMemoryPool *pmp, CExpression *pexpr, DrgPcrs *pdrgpcrsFst, DrgPcrs *pdrgpcrsSnd);
+			CColRefSetArray *PdrgpcrsMergeFromBoolOp(IMemoryPool *mp, CExpression *pexpr, CColRefSetArray *pdrgpcrsFst, CColRefSetArray *pdrgpcrsSnd);
```
Comment (sambitesh) [inline libgpopt/include/gpopt/base/CConstraint.h]: can we replace pdrgpcrsFst etc too?

File: libgpopt/include/gpopt/base/CConstraintInterval.h [hunk 7]
```diff
@@ -51,9 +51,9 @@ namespace gpopt
 			const CColRef *m_pcr;
 
 			// array of ranges
-			DrgPrng *m_pdrgprng;
+			CRangeArray *m_pdrgprng;
 
-			// does the interval include the null value
+			// does the interval include the null m_bytearray_value
```
Comment (sambitesh) [inline libgpopt/include/gpopt/base/CConstraintInterval.h]: m_bytearray_value to be replaced by value ?

File: libgpopt/include/gpopt/base/CConstraintInterval.h [hunk 8]
```diff
@@ -183,12 +183,12 @@ namespace gpopt
...
 
 			// all ranges in interval
-			DrgPrng *Pdrgprng() const
+			CRangeArray *Pdrgprng() const
 			{
 				return m_pdrgprng;
 			}
 
-			// does the interval include the null value
+			// does the interval include the null m_bytearray_value
```
Comment (sambitesh) [inline libgpopt/include/gpopt/base/CConstraintInterval.h]: value?

File: libgpopt/include/gpopt/base/CCostContext.h [hunk 9]
```diff
@@ -204,7 +204,7 @@ namespace gpopt
 				return m_pdpplan;
 			}
 
-			// set cost value
+			// set cost m_bytearray_value
```
Comment (sambitesh) [inline libgpopt/include/gpopt/base/CCostContext.h]: value

File: libgpopt/include/gpopt/base/CDrvdPropScalar.h [hunk 10]
```diff
@@ -64,7 +64,7 @@ namespace gpopt
 			// scalar expression contains non-scalar function?
 			BOOL m_fHasNonScalarFunction;
 
-			// total number of Distinct Aggs (e.g., {count(distinct a), sum(distinct a), count(distinct b)}, the value is 3),
+			// total number of Distinct Aggs (e.g., {count(distinct a), sum(distinct a), count(distinct b)}, the m_bytearray_value is 3),
```
Comment (sambitesh) [inline libgpopt/include/gpopt/base/CDrvdPropScalar.h]: value

File: libgpopt/include/gpopt/base/CMaxCard.h [hunk 11]
```diff
@@ -41,7 +41,7 @@ namespace gpopt
 	
 		private:
 
-			// actual (cropped) value
+			// actual (cropped) m_bytearray_value
```
Comment (sambitesh) [inline libgpopt/include/gpopt/base/CMaxCard.h]: value

File: libgpopt/include/gpopt/cost/ICostModelParams.h [hunk 12]
```diff
@@ -52,32 +52,32 @@ namespace gpopt
 				private:
 
 					// param identifier
-					ULONG m_ulId;
+					ULONG m_id;
 
-					// param value
-					CDouble m_dVal;
+					// param m_bytearray_value
```
Comment (sambitesh) [inline libgpopt/include/gpopt/cost/ICostModelParams.h]: value

**Comment by khannaekta** [comment]
Pushed changes for @sambitesh's review comment on `m_bytearray_value`

File: StyleGuide.md [hunk 13]
```diff
@@ -0,0 +1,84 @@
...
+found [here](https://google.github.io/styleguide/cppguide.html); with some
+modifications that are enumerated in this document.
+
+## Guiding Principles
+
+1. Although ORCA is maintained in a separate repository, a lot of development
+	 on ORCA is closely related work in GPDB. To reduce context switch costs
+	 arising from such a such a development model, we would like to keep the
+	 style in ORCA as close as possible to GPDB. This includes using tabs instead
+	 of spaces for indentation, all cap global constants etc.
```
Comment (sambitesh) [inline StyleGuide.md]: Would it be better to use Greenplum instead of `GPDB` here ?
@hardikar please let us know if you agree and we can make this change.

**Pull request merged**